app/scripts/services/serviceInstances.js

```diff
@@ -2,65 +2,129 @@
 
 angular.module("openshiftConsole")
   .factory("ServiceInstancesService", function($filter,
+                                               $q,
                                                $uibModal,
+                                               BindingService,
+                                               CatalogService,
                                                DataService,
+                                               Logger,
                                                NotificationsService) {
+    var getBindingsIfNecessary = function(apiObject, bindings) {
+      var deferred = $q.defer();
 
-    function deprovision(apiObject) {
-      var getErrorDetails = $filter('getErrorDetails');
-      var modalScope = {
-        alerts: {
-          deprovision: {
-            type: 'error',
-            message: 'Service \'' + apiObject.spec.serviceClassName + '\' will be deleted and no longer available.'
+      if (angular.isDefined(bindings)) {
+        return $q.when(bindings);
+      } else {
+        var context = {namespace: apiObject.metadata.namespace};
+        var resource = {
+          group: 'servicecatalog.k8s.io',
+          resource: 'serviceinstancecredentials'
+        };
+
+        DataService.list(resource, context, function (serviceBindings) {
+          bindings = serviceBindings.by('metadata.name');
+          bindings = BindingService.getBindingsForResource(bindings, apiObject);
+          deferred.resolve(bindings);
+        });
+      }
+
+      return deferred.promise;
```

Comment:
You can just return the promise from `DataService.list` instead of creating your own deferred.

```js
    return DataService.list(resource, context, function (serviceBindings) {
      bindings = serviceBindings.by('metadata.name');
      return BindingService.getBindingsForResource(bindings, apiObject);
    });
```

Reply:
@spadgett What comes back from the promise is not the bindings manipulated by the callback function. Maybe I'm doing something else wrong?

Reply:
Sorry my JS was wrong:

```js
   return DataService.list(resource, context).then(function(serviceBindings) {
      bindings = serviceBindings.by('metadata.name');
      return BindingService.getBindingsForResource(bindings, apiObject);
    });
```

Reply:
Ah yes, that makes sense. Sorry, I should have seen that.

Reply:
Fixed. Thanks @spadgett